test/core/networking/server_test.dart

```diff
@@ -0,0 +1,296 @@
+
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      wsOnError: (ServerSession sess, Object error) {
+        called.set(error);
+      },
+      wsOnMessage: (ServerSession sess, dynamic msg) async {
+        sess.raise(TestException(msg));
+      },
+    );
+    await server.ready;
+
+    // Connect client
+    final ClientSession client = ClientSession(getUri(server, 'ws').toString());
+    client.send('ANYTHING');
+
+    // Wait for handler to be called
+    await called.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    called.expect(
+      isA<TestException>().having(
+        (TestException e) => e.message,
+        'message',
+        equals('ANYTHING'),
+      ),
+    );
+
+    // Teardown
+    client.close();
+    server.close();
+  });
+  test('OnDone is called', () async {
+    final VHook<bool?> called = VHook<bool?>(null);
+
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      wsOnDone: (ServerSession sess) {
+        called.set(true);
+      },
+      wsOnMessage: (ServerSession sess, dynamic msg) async {},
+    );
+    await server.ready;
+
+    // Connect client
+    final ClientSession client = ClientSession(getUri(server, 'ws').toString());
+    await client.send('ANYTHING');
+    client.close();
+
+    // Wait for handler to be called
+    await called.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    called.expect(equals(true));
+
+    // Teardown
+    client.close();
+    server.close();
+  });
+  test('Get address', () async {
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+    );
+    await server.ready;
+
+    expect(server.address, equals(InternetAddress.loopbackIPv4));
+
+    // Teardown
+    server.close();
+  });
+  test('Get port', () async {
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+    );
+    await server.ready;
+
+    expect(server.port, equals(isA<int>()));
+
+    // Teardown
+    server.close();
+  });
+  test('Get isEmpty', () async {
+    final VHook<bool?> received = VHook<bool?>(null);
+
+    WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      wsOnMessage: (ServerSession sess, dynamic msg) async {
+        received.set(true);
+      },
+    );
+    await server.ready;
+
+    expect(server.isEmpty, equals(true));
+
+    // Connect a client
+    final ClientSession client = ClientSession(getUri(server, 'ws').toString());
+    client.send('ANYTHING');
+
+    // Check if server is still empty
+    await received.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    received.expect(equals(true));
+
+    expect(server.isEmpty, equals(false));
+
+    // Teardown
+    client.close();
+    await server.close();
+
+    // Is empty again
+    expect(server.isEmpty, equals(true));
+  });
+}
+
+Uri getUri(WebServer? server, String prefix) {
```

Comment:
```suggestion
Uri _getUri(WebServer? server, String prefix) {
```

Reply:
and some docs please

Reply:
Why should they be private? Test files will **never** be imported? They even have their own __main__.

Reply:
Because it's confusing if you make possible exportable file members if you don't want to export them. You also don't make every method public just because you know you should not use it somewhere else, do you?